compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -12810,6 +12810,149 @@ TR::Node *su2dSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier * s)
+         newConst = isAddOp ? (oldConst2 - oldConst1) : (oldConst2 + oldConst1);
+
+         // Get type ranges to check for overflow or underflow
+         int64_t signedMax = 0, signedMin = 0, unsignedMax = 0, unsignedMin = 0;
+         if (opNode->getOpCode().is1Byte())
+            {
+            signedMax   = TR::getMaxSigned<TR::Int8>();
+            signedMin   = TR::getMinSigned<TR::Int8>();
+            unsignedMax = TR::getMaxUnsigned<TR::Int8>();
+            unsignedMin = TR::getMinUnsigned<TR::Int8>();
+            }
+         else if (opNode->getOpCode().is2Byte())
+            {
+            signedMax   = TR::getMaxSigned<TR::Int16>();
+            signedMin   = TR::getMinSigned<TR::Int16>();
+            unsignedMax = TR::getMaxUnsigned<TR::Int16>();
+            unsignedMin = TR::getMinUnsigned<TR::Int16>();
+            }
+         else if (opNode->getOpCode().is4Byte())
+            {
+            signedMax   = TR::getMaxSigned<TR::Int32>();
+            signedMin   = TR::getMinSigned<TR::Int32>();
+            unsignedMax = TR::getMaxUnsigned<TR::Int32>();
+            unsignedMin = TR::getMinUnsigned<TR::Int32>();
+            }
+         else
+            return node;
+
+         if ((!isUnsignedCompare && (newConst > signedMax || newConst < signedMin)) ||
+                 (isUnsignedCompare && (newConst > unsignedMax || newConst < unsignedMin)))
+            return node;
+         }
+      else  // Long
+         {
+         if (isUnsignedCompare)
+            {
+            uint64_t oldUConst1 = opNode->getSecondChild()->getUnsignedLongInt();
+            uint64_t oldUConst2 = secondChild->getUnsignedLongInt();
+
+            if (isAddOp && (oldUConst2 > oldUConst1))
+               newUConst = oldUConst2 - oldUConst1;
+            else if (isSubOp && (oldUConst2 <= TR::getMaxUnsigned<TR::Int64>() - oldUConst1))
+               newUConst = oldUConst2 + oldUConst1;
+            else
+               return node;
+            }
+         else // signed long compares
+            {
+            oldConst1 = opNode->getSecondChild()->getConstValue();
+            oldConst2 = secondChild->getConstValue();
+
+            if (isAddOp &&
+                    !(oldConst1 > 0 && (oldConst2 < TR::getMinSigned<TR::Int64>() + oldConst1)) &&
+                    !(oldConst1 < 0 && (oldConst2 > TR::getMaxSigned<TR::Int64>() + oldConst1)))
+               newConst = oldConst2 - oldConst1;
+            else if (isSubOp &&
+                     !(oldConst1 > 0 && (oldConst2 > TR::getMaxSigned<TR::Int64>() - oldConst1)) &&
+                     !(oldConst1 < 0 && (oldConst2 < TR::getMinSigned<TR::Int64>() - oldConst1)))
+               newConst = oldConst2 + oldConst1;
+            else
+               return node;
+            }
+         }
+
+      // Done checking constant values. Transform the tree.
+      performTransformation(s->comp(), "%sEliminating add/sub operation under integer comparison node n%dn\n",
```

Comment:
this needs to be in an if so it can be diabled

Reply:
The function `removeArithmeticsUnderIntegralCompare` will return right away if this simplification is disabled. I don't think we need another `if` around this line.

Reply:
The purpose of `performTransformation` is to allow disabling a particular occurrence of a transformation without disabling the class of transformation, e.g. so that it's possible to debug with `lastOptSubIndex`

Reply:
+1 the perform transformation needs to be the condition of an if statement guarding the rest of the code.